hicommonwealth/commonwealth
# New topic flow - FE & BE integration

## Link to Issue
Closes: #9224 

## Description of Changes
- wired up frontend flow and forms to backend mutations and queries

## Test Plan
- the whole flow described [here ](https://www.figma.com/board/xsXNa51D2oFNjOpXr5AveR/Weighted-Voting-topic-creation-flow?node-id=0-1&node-type=canvas&t=IwlxCSsft9czGgyd-0)should be testable now
- enable flag FLAG_FARCASTER_CONTEST=true
- create community (you should not be able to enable stake in create community flow anymore)

### 1. Create regular topic
- go to `topics` menu
- fill out the form and click `next`
- click `no, create regular topic`
- you should be redirected to the topic page

https://github.com/user-attachments/assets/85ac22f7-5a92-4128-8fe2-120aebc72037


### 2. Create erc20 topic
- go to `topics` menu
- fill out the form and click `next`
- click `yes` to enable weighted voting
- select `connect ERC20 token` button and click `continue`
- fill out ERC20 form (eg use this address 0x19848077f45356b21164c412eff3d3e4ff6ebc31)
- click `enable weighted voting for topic` 
- you should be redirected to the topic page and you should see the banner on the top of the page with token information

https://github.com/user-attachments/assets/7a289e9d-aab8-4dad-9595-0c705aa3c55d

### 3. Create stake topic if stake is not enabled
- go to `topics` menu
- fill out the form and click `next`
- click `yes` to enable weighted voting
- select `use community stake` button and click `continue`
- enable stake by signing two transactions
- on success you should see the page with stake details 
- click `Create topic with stakes` 
- you should be redirected to the topic page

https://github.com/user-attachments/assets/2f3f2188-5298-4de6-999a-1dfb20f14228

### 4. Create stake topic if stake is enabled
- this is the same as numer 3 ☝️ but you don't need to sign transactions

## Deployment Plan

1. n/a

## Other Considerations

- n/a
